tests/func-tests/nonroot_root_test.go

```diff
@@ -0,0 +1,138 @@
+package tests_test
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/pointer"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+	"kubevirt.io/client-go/kubecli"
+)
+
+var _ = Describe("nonRoot -> root FG", func() {
+	tests.FlagParse()
+	var cli kubecli.KubevirtClient
+	ctx := context.TODO()
+
+	cli, err := kubecli.GetKubevirtClient()
+	Expect(cli).ToNot(BeNil())
+	Expect(err).ToNot(HaveOccurred())
+	var initialNonRoot *bool
+	var initialRoot *bool
+
+	el := field.ErrorList{field.Invalid(field.NewPath("spec", "featureGates"), "object", "nonRoot FG is deprecated, please use root FG with opposite logic")}
+	invalidHCOError := errors.NewInvalid(schema.GroupKind{Kind: util.HyperConvergedKind, Group: util.APIVersionGroup}, util.HyperConvergedName, el)
+	invalidHCOError.ErrStatus.APIVersion = "v1"
+	invalidHCOError.ErrStatus.Kind = "Status"
+
+	BeforeEach(func() {
+		tests.BeforeEach()
+		hc := tests.GetHCO(ctx, cli)
+		initialNonRoot = hc.Spec.FeatureGates.NonRoot
+		initialRoot = hc.Spec.FeatureGates.Root
+	})
+
+	AfterEach(func() {
+		hc := tests.GetHCO(ctx, cli)
+		hc.Spec.FeatureGates.NonRoot = initialNonRoot
+		hc.Spec.FeatureGates.Root = initialRoot
+		hc = tests.UpdateHCORetry(ctx, cli, hc)
+	})
+
+	DescribeTable("should correctly handle nonRoot -> root FG transition",
+		func(nonRootFG *bool, rootFG *bool, expectedErr error, expectedNonRootFG *bool, expectedRootFG *bool) {
+			if expectedErr == nil {
+				hc := tests.GetHCO(ctx, cli)
+				hc.Spec.FeatureGates.NonRoot = nonRootFG
+				hc.Spec.FeatureGates.Root = rootFG
+				hc = tests.UpdateHCORetry(ctx, cli, hc)
+				Expect(hc.Spec.FeatureGates.NonRoot).To(Equal(expectedNonRootFG))
+				Expect(hc.Spec.FeatureGates.Root).To(Equal(expectedRootFG))
+			} else {
+				Eventually(func() error {
+					hc := tests.GetHCO(ctx, cli)
+					hc.Spec.FeatureGates.NonRoot = nonRootFG
+					hc.Spec.FeatureGates.Root = rootFG
+					hc, err = tests.UpdateHCO(ctx, cli, hc)
+					return err
```

Comment:
Do you want to check the KubeVirt CR as well?

Reply:
done